packages/tupaia-web-server/src/routes/ReportRoute.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ *
+ */
+
+import { Request } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+
+export type ReportRequest = Request<
+  { reportCode: string },
+  any,
+  any,
+  any
+>;
+
+export class ReportRoute extends Route<ReportRequest> {
+  public async buildResponse() {
+    const { query, ctx } = this.req;
+    const { reportCode } = this.req.params;
+    const { legacy } = query;
```

Comment:
Out of curiosity, are we initially leaving the `tupaia-web <-> tupaia-web-server` interface as is? Then perhaps reworking at a later point?

It always struck me as a little odd that the front-end told the backend if the report were legacy or not...

Reply:
Currently I'm just making it backwards compatible, because that's the easiest way to test backend before we've updated the frontend. We can definitely make adjustments to the API during this project, but I reckon better to strive for a "cleaned up" version of the current API rather than redesigning too much.

It'd be a little annoying to work out if the report is legacy in `tupaia-web-server`, since as an orchestration server it doesn't have direct DB access, so would need to do a couple of api calls to figure it out. And I think there's not currently a rule that says that a `legacy_report` and a `report` can't have the same `code`. So we can assume but not guarantee (or more realistically we'd just add that rule)

Splitting the difference, since the frontend already knows whether or not it's legacy maybe we could split into separate routes `/api/report/:code` and `/api/legacyReport/:code`

Reply:
I like having 2 routes. 👍